lib/preview/link.js

```diff
@@ -102,18 +102,41 @@ var getYoutubeId = function(link) {
  */
 var processYoutube = function(ctx, youtubeId, callback) {
   var thumbnailUrl = util.format('http://img.youtube.com/vi/%s/hqdefault.jpg', youtubeId);
-  CoreUtil.download(ctx, thumbnailUrl, 'thumbnail.jpq', function(err, thumbnailPath, mimeType) {
+  CoreUtil.download(ctx, thumbnailUrl, 'youtube_thumbnail.jpg', function(err, thumbnailDownloadPath, mimeType) {
     if (err) {
       return callback(err);
     } else if (mimeType !== 'image/jpeg') {
-      return callback({'code': 500, 'msg': 'Unexpected mime type for a youtube thumbnail'});
+      return callback({'code': 500, 'url': thumbnailUrl, 'msg': 'Unexpected mime type for a YouTube thumbnail image'});
     }
 
-    var metadata = {
-      'youtubeId': youtubeId
-    };
-    var result = new Result(PreviewConstants.STATUS.DONE, thumbnailPath, null, null, metadata);
-    return callback(null, result);
+    PreviewImage.generateThumbnail(ctx, thumbnailDownloadPath, function(err, thumbnailPath) {
+      if (err) {
+        return callback(err);
+      }
+
+      var previewUrl = util.format('http://img.youtube.com/vi/%s/maxresdefault.jpg', youtubeId);
```

Comment:
@paulkerschen will this "max res" preview URL work in call cases? I do not get our beloved Winona + pizza with https://img.youtube.com/vi/mzDwKMgsfmo/maxresdefault.jpg BUT the mqdefault URL works: https://img.youtube.com/vi/mzDwKMgsfmo/mqdefault.jpg  

Expected behavior:
![image](https://cloud.githubusercontent.com/assets/7606442/23522452/e45f8166-ff37-11e6-9c0a-bca151deeb40.png)